### PR TITLE
Wireless screen share

### PR DESCRIPTION
**Pull request by arun3528**
# Pull Request Template
Meeting plugin can share and receive share.  
Join method currently only joins the meeting . addMedia function adds the actual media to the meeting 
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Test Coverage

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* SDK Version
* Node/Browser Version
* NPM Version

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/node_modules/@webex/plugin-meetings/src/constants.js [hunk 1]
```diff
@@ -128,7 +129,8 @@ export const EVENT_TYPES = {
   OTHER: 'other',
   LOCAL: 'local',
   REMOTE: 'remote',
-  SCREEN: 'screen',
+  REMOTE_SHARE: 'local-share',
```
Comment (arun3528) [inline packages/node_modules/@webex/plugin-meetings/src/constants.js]: camel case

File: packages/node_modules/@webex/plugin-meetings/src/media/index.js [hunk 2]
```diff
@@ -61,14 +61,30 @@ Media.generateLocalMedias = (mediaId, audio, video) => {
...
+    return Media.getDisplayMedia(
+      {
+        sendAudio: false, // TODO : will change in future
+        sendShare: true
+      }
+    );
+  }
+
+  return Promise.resolve(undefined);
 };
```
Comment (arun3528) [inline packages/node_modules/@webex/plugin-meetings/src/media/index.js]: js docs

File: packages/node_modules/@webex/plugin-meetings/src/meeting/index.js [hunk 3]
```diff
@@ -366,11 +408,14 @@ export default class Meeting extends StatelessSparkPlugin {
    * @returns {Promise}
    * @memberof Meeting
    */
-  join(options = {}) {
+  joinWithMedia(options = {}) {
```
Comment (arun3528) [inline packages/node_modules/@webex/plugin-meetings/src/meeting/index.js]: remove

File: packages/node_modules/@webex/plugin-meetings/src/meeting/index.js [hunk 4]
```diff
@@ -379,30 +424,115 @@ export default class Meeting extends StatelessSparkPlugin {
...
+   * @param {object} options A configurable options object for joining a meeting
+   * @param {Object } options.resourceId pass the deviceId
+   * @returns {Promise}
+   * @memberof Meeting
+   */
+  addMedia(options = {}) {
+    const {localStream, localShare} = options;
+    // eslint-disable-next-line no-warning-comments
+    // TODO wire into default config. There's currently an issue with the stateless plugin or how we register
+    const mediaDirection = Object.assign(this.config.mediaSettings, options.mediaSettings);
```
Comment (arun3528) [inline packages/node_modules/@webex/plugin-meetings/src/meeting/index.js]: add setup function

File: packages/node_modules/@webex/plugin-meetings/src/meeting/index.js [hunk 5]
```diff
@@ -521,16 +652,23 @@ export default class Meeting extends StatelessSparkPlugin {
...
+   *
+   * Alias of updateShare
+   * @param {*} args
+   * @returns {updateShare}
+   * @memberof Meeting
+   */
+  share() {
+    const content = this.locus.mediaShares.find((element) => element.name === 'content');
+    this.meetingRequest.changeMeetingFloor({
+      disposition: 'GRANTED', // : 'RELEASED';
```
Comment (arun3528) [inline packages/node_modules/@webex/plugin-meetings/src/meeting/index.js]: add constants

File: packages/node_modules/@webex/plugin-meetings/src/meeting/request.js [hunk 6]
```diff
@@ -45,6 +45,12 @@ export default class MeetingRequest extends StatelessSparkPlugin {
       };
     }
 
+    if (resourceId === sipUri) {
+      body.callPreferences = {
+        requestedMedia: ['SLIDES']
```
Comment (arun3528) [inline packages/node_modules/@webex/plugin-meetings/src/meeting/request.js]: constants

File: packages/node_modules/@webex/plugin-meetings/src/meeting/request.js [hunk 7]
```diff
@@ -100,4 +106,30 @@ export default class MeetingRequest extends StatelessSparkPlugin {
       body
     });
   }
+
+  changeMeetingFloor(options) {
+    let floorReq = {disposition: options.disposition};
+    /* istanbul ignore else */
+    if (options.disposition === 'GRANTED') {
```
Comment (arun3528) [inline packages/node_modules/@webex/plugin-meetings/src/meeting/request.js]: constants

File: packages/node_modules/@webex/plugin-meetings/src/meeting/request.js [hunk 8]
```diff
@@ -100,4 +106,30 @@ export default class MeetingRequest extends StatelessSparkPlugin {
...
+        disposition: options.disposition,
+        requester: {
+          url: options.personUrl
+        }
+      };
+    }
+
+    return this.request({
+      uri: options.uri,
+      method: 'PUT',
```
Comment (arun3528) [inline packages/node_modules/@webex/plugin-meetings/src/meeting/request.js]: contansts

File: packages/node_modules/@webex/plugin-meetings/src/meeting/index.js (hunk 3 above)
Comment (arun3528) [reply packages/node_modules/@webex/plugin-meetings/src/meeting/index.js]: done

File: packages/node_modules/@webex/plugin-meetings/sample/index.html [hunk 9]
```diff
@@ -116,14 +228,15 @@ <h2>Then click:</h2>
...
     </form>
     <div>
-      <video style="max-height: 100px, max-width: 100px" id="remotevideo" autoplay></video>
-      <video style="max-height: 100px, max-width: 100px" id="screenvideo" autoplay></video>
-      <video style="max-height: 100px, max-width: 100px" id="localvideo" muted="true" autoplay></video>
-      <video style="max-height: 100px, max-width: 100px" id="localscreen" autoplay></video>
+      <video style="--width: 20%;width: var(--width);height: calc(var(--width) * 0.75);" id="localvideo" muted="true" autoplay></video>
+      <video style="--width: 35%;width: var(--width);height: calc(var(--width) * 0.75);" id="remotevideo" autoplay></video>
+      <video style="--width: 35%;width: var(--width);height: calc(var(--width) * 0.75);" id="sharevideo" autoplay></video>
+      <video style="--width: 20%;width: var(--width);height: calc(var(--width) * 0.75);" id="localshare" autoplay></video>
```
Comment (andrewholsted) [inline packages/node_modules/@webex/plugin-meetings/sample/index.html]: could probably have moved these styles up to the style block you added but it's fine

File: packages/node_modules/@webex/plugin-meetings/src/constants.js [hunk 10]
```diff
@@ -128,7 +134,8 @@ export const EVENT_TYPES = {
   OTHER: 'other',
   LOCAL: 'local',
   REMOTE: 'remote',
-  SCREEN: 'screen',
+  REMOTE_SHARE: 'localShare',
```
Comment (andrewholsted) [inline packages/node_modules/@webex/plugin-meetings/src/constants.js]: 👍

File: packages/node_modules/@webex/plugin-meetings/src/media/index.js [hunk 11]
```diff
@@ -61,14 +61,37 @@ Media.generateLocalMedias = (mediaId, audio, video) => {
...
   // we return undefined here as the second param until screen share is implemented
-  return Promise.resolve().then(() => Media.getMedia(sendAudio, sendVideo).then((media) => [media, undefined]));
-};
 
-Media.addMedia = (mediaDirection, pcMgr, audioVideo, screen) => {
+  if (sendAudio || sendVideo) {
+    return Media.getMedia(sendAudio, sendVideo);
+  }
+
+  if (sendShare) {
```
Comment (andrewholsted) [inline packages/node_modules/@webex/plugin-meetings/src/media/index.js]: Is it not possible to do both audio/video and share at the same time? Or is that just handled elsewhere?

File: packages/node_modules/@webex/plugin-meetings/src/meeting/index.js [hunk 12]
```diff
@@ -121,12 +121,8 @@ export default class Meeting extends StatelessSparkPlugin {
    * @returns {null}
    */
   setSipUri(sipUri) {
-    if (this.meetingInfo) {
-      this.setMeetingInfo(this.meetingInfo);
-    }
-    else if (!this.sipUri) {
-      this.sipUri = sipUri;
-    }
+    // This can be tel no, device id or a sip uri, user Id
```
Comment (andrewholsted) [inline packages/node_modules/@webex/plugin-meetings/src/meeting/index.js]: Maybe a better var name is warranted then?

File: packages/node_modules/@webex/plugin-meetings/src/meetings/index.js [hunk 13]
```diff
@@ -99,8 +99,11 @@ export default class Meetings extends SparkPlugin {
       }
     );
     // TODO: code review check: that said, 1:1 will always be done with sipUri, and setMeetingInfo will overwrrite it if info exists as necessary
-    meeting.setMeetingInfo(mtgInfo);
-    meeting.setSipUri(destination);
+    meeting.parseMeetingInfo(mtgInfo);
+    if (!meeting.sipUri) {
+      meeting.setSipUri(destination);
+    }
```
Comment (andrewholsted) [inline packages/node_modules/@webex/plugin-meetings/src/meetings/index.js]: 👍

**Pull request merged**